src/lazyoperations.jl

```diff
@@ -59,6 +100,181 @@ function tr(K::Kron{<:Any, 2})
 end
 
 
+function diag(K::Kron{<:Any, 2})
+    if all(A -> (size(A, 1) == size(A, 2)), K.args)  # check if all component matrices are square
+        return kron(diag.(K.args)...)
+    else
+        d = similar(K.args[1], minimum(size(K)))
+        for i in 1:length(d)
+            d[i] = K[i,i]
+        end
+        return d
+    end
+end
+
+
+function LazyArrays.materialize(M::Applied{
```

Comment:
No need for `LazyArrays.` here and below

Reply:
oops! this was a hold-over from when i was playing around with this in jupyter :joy: